s/api/services/v1/comment-writing-service.ts

```diff
@@ -102,41 +102,43 @@ export const makeCommentWritingService = asServiceProvider(({
 		})
 	},
 
-	async archiveComment(rawId: string): Promise<void> {
-		const id = enforceValidation(rawId, validateId)
+	async archiveComments(rawIds: string[]): Promise<void> {
+		const idArr = enforceValidation(rawIds, validateIdArray)
 
 		if (!user)
 			throw new renraku.ApiError(403, "cannot archive, not logged in")
 
-		const specificComment = await database.tables.comments.readOne(
-			find({id: dbmage.Id.fromString(id)})
-		)
-
-		if (!specificComment)
-			throw new renraku.ApiError(404, "cannot archive, comment not found")
-
-		const userIsTheAuthor = user.id === specificComment.authorId.string
-		const userHasAdminRights = user.permissions.canArchiveAnyComment
-		const userIsAllowedToArchive = userIsTheAuthor || userHasAdminRights
-
-		if (!userIsAllowedToArchive) 
-			throw new renraku.ApiError(403, "forbidden, must be author or admin to archive a comment")
-
-		await database.transaction(async({tables}) => {
-			await tables.comments.update({
-				...find({id: dbmage.Id.fromString(id)}),
-				write: {archived: true},
-			})
-
-			// TODO dbmage currently requires this count.. but shouldn't
-			const scoreCount = await tables.scores.count({
-				...find({commentId: dbmage.Id.fromString(id)}),
-			})
-			if (scoreCount)
-				await tables.scores.update({
-					...find({commentId: dbmage.Id.fromString(id)}),
+		for (let index = 0; index < idArr.length; index++) {
+			const id = idArr[index];
+			const specificComment = await database.tables.comments.readOne(
+				find({id: dbmage.Id.fromString(id)})
+			)
+			if (!specificComment)
+				throw new renraku.ApiError(404, "cannot archive, comment not found")
+
+				const userHasAdminRights = user.permissions.canArchiveAnyComment
+				const userIsAllowedToArchive = userHasAdminRights
+		
+			if (!userIsAllowedToArchive) 
+				throw new renraku.ApiError(403, "forbidden, must be author or admin to archive a comment")
+
+			await database.transaction(async({tables}) => {
+				await tables.comments.update({
+					...find({id: dbmage.Id.fromString(id)}),
```

Comment:
we really need to run a single database call, using dbmage.findAll, to archive all the comments in one call